js/src/Components/Widgets/LinkWidget.js

```diff
@@ -1,20 +1,44 @@
 import React, {useState} from 'react';
 import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
 import FormGroup from '@material-ui/core/FormGroup';
 import FormControl from '@material-ui/core/FormControl';
 import FormLabel from '@material-ui/core/FormLabel';
 import {FormHelperText} from "@material-ui/core";
 import Autocomplete from '@material-ui/lab/Autocomplete';
 
 export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) => {
-
   let timeout;
+  let initialCondition = defaultValue;
+  if (initialCondition === undefined){
+    initialCondition = [];
+    initialCondition.push(emptyLinkValue());
+  }
+
   const [urlSuggestions, setSuggestions] = useState([]);
+  const [fieldValues, setValues] = useState(initialCondition);
 
-  const defaultFieldValue = {
-    uri: defaultValue && defaultValue.length ? defaultValue[0].uri : '',
-    title: defaultValue && defaultValue.length ? defaultValue[0].title : ''
-  };
+  function emptyLinkValue() {
+    return {
+      uri: '',
+      title: '',
+      options: [],
+      target_uuid: false
+    };
+  }
+
+  const alterValues = (props) => {
+    const newState = Array.from(fieldValues);
+    newState[props.delta].title = props.title;
+    newState[props.delta].uri = props.uri;
+    setValues(newState);
+    onFieldChange(fieldValues);
```

Comment:
why do `setValues` and `onFieldChange` use different variables?

Reply:
So, in this case, `setValues` updates the local state, which was set up via the `useState` hook in line 19.  `fieldValues` is the current local state.  `onFieldChange` is the function that is passed in via the constructor, which handles the state one layer up, in the `ParagraphForm.js`, and does the work of setting the field in Drupal, right?

So I created the local state so I could change it, since the widget is declared fully functionally, rather than object oriented, otherwise I couldn't change it inside the widget.  When the local state is altered, I feed that back up into the `onFieldChange` function so that it'll persist outside of the widget.

Reply:
This is now fixed to use only the onFieldChange function.